sycl/include/CL/__spirv/spirv_ops.hpp

```diff
@@ -136,100 +160,374 @@ __SPIRV_ATOMICS(__SPIRV_ATOMIC_UNSIGNED, unsigned long long)
+                                dataT Data) noexcept;
 
 template <typename dataT>
-extern dataT
+extern SYCL_EXTERNAL dataT
 __spirv_SubgroupBlockReadINTEL(const __attribute__((opencl_global)) uint32_t *Ptr) noexcept;
 
 template <typename dataT>
-extern void __spirv_SubgroupBlockWriteINTEL(__attribute__((opencl_global)) uint32_t *Ptr,
-                                            dataT Data) noexcept;
+extern SYCL_EXTERNAL void
+__spirv_SubgroupBlockWriteINTEL(__attribute__((opencl_global)) uint32_t *Ptr,
+                                dataT Data) noexcept;
 
 template <typename dataT>
-extern int32_t __spirv_ReadPipe(RPipeTy<dataT> Pipe, dataT *Data,
-                                int32_t Size, int32_t Alignment) noexcept;
+extern SYCL_EXTERNAL int32_t
+__spirv_ReadPipe(RPipeTy<dataT> Pipe, dataT *Data, int32_t Size,
+                 int32_t Alignment) noexcept;
 template <typename dataT>
-extern int32_t __spirv_WritePipe(WPipeTy<dataT> Pipe, const dataT *Data,
-                                 int32_t Size, int32_t Alignment) noexcept;
+extern SYCL_EXTERNAL int32_t
+__spirv_WritePipe(WPipeTy<dataT> Pipe, const dataT *Data, int32_t Size,
+                  int32_t Alignment) noexcept;
 template <typename dataT>
-extern void __spirv_ReadPipeBlockingINTEL(RPipeTy<dataT> Pipe, dataT *Data,
-                                          int32_t Size,
-                                          int32_t Alignment) noexcept;
+extern SYCL_EXTERNAL void
+__spirv_ReadPipeBlockingINTEL(RPipeTy<dataT> Pipe, dataT *Data, int32_t Size,
+                              int32_t Alignment) noexcept;
 template <typename dataT>
-extern void __spirv_WritePipeBlockingINTEL(WPipeTy<dataT> Pipe,
-                                           const dataT *Data, int32_t Size,
-                                           int32_t Alignment) noexcept;
+extern SYCL_EXTERNAL void
+__spirv_WritePipeBlockingINTEL(WPipeTy<dataT> Pipe, const dataT *Data,
+                               int32_t Size, int32_t Alignment) noexcept;
 template <typename dataT>
-extern RPipeTy<dataT> __spirv_CreatePipeFromPipeStorage_read(
+extern SYCL_EXTERNAL RPipeTy<dataT> __spirv_CreatePipeFromPipeStorage_read(
     const ConstantPipeStorage *Storage) noexcept;
 template <typename dataT>
-extern WPipeTy<dataT> __spirv_CreatePipeFromPipeStorage_write(
+extern SYCL_EXTERNAL WPipeTy<dataT> __spirv_CreatePipeFromPipeStorage_write(
     const ConstantPipeStorage *Storage) noexcept;
 
-extern void __spirv_ocl_prefetch(const __attribute__((opencl_global)) char *Ptr,
-                                 size_t NumBytes) noexcept;
+extern SYCL_EXTERNAL void
+__spirv_ocl_prefetch(const __attribute__((opencl_global)) char *Ptr,
+                     size_t NumBytes) noexcept;
+
+// __attribute((format(...))) enables compiler checks for a format string.
+SYCL_EXTERNAL
+int __spirv_ocl_printf(const __attribute__((opencl_constant)) char *fmt, ...)
+    __attribute__((format(printf, 1, 2)));
```

Comment:
`format` attribute here might cause undesired side-effects, see [this](https://github.com/intel/llvm/pull/835#discussion_r361439440) comment

Reply:
Also, please remove similar declaration from `sycl/include/CL/sycl/intel/builtins.hpp`

Reply:
> format attribute here might cause undesired side-effects, see this comment

Somehow everyone was OK with this attribute residing behind __spirv_ocl_printf decl in sycl/source/detail/devicelib/fallback-cassert.cpp

Reply:
It is obviously wrong to use the normal printf format warning. But why can't we use the OCL printf warning? It seems it is (at least partly) implemented: http://llvm.org/viewvc/llvm-project?view=revision&revision=352540. A partly implemented warning (as long as it incomplete and doesn't give false positives) is better than none.